src/Integration/Telemetry/TelemetryManager.cs

```diff
@@ -248,16 +251,13 @@ public void SecretDetected(string ruleId)
         {
             var rulesUsage = telemetryRepository.Data.RulesUsage;
 
-            if (!rulesUsage.RulesThatRaisedIssues.Contains(ruleId))
+            lock (Lock)
```

Comment:
Similar to the issue above, List.Contains shouldn't be called outside of the lock if another thread can call List.Add in the meanwhile. For readability, it is easier to just wrap everything in a Lock, as that also removes the need for Distinct.